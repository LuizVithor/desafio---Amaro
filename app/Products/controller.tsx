import { product } from "@/common/components/Products/controller";

export interface responseGetProducts {
    skip: number;
    total: number;
    limit: number;
    products: product[];
}