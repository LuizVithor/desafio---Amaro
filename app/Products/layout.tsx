import { ReactNode } from "react"
import Sidebar from "@/common/components/Products/SideBar"

const ProductsLayout = ({ children }: { children: ReactNode }) => {
    return (
        <>
            <Sidebar />
            <div
                style={{
                    width: "100%",
                    height: "100%",
                    paddingTop: "64px"
                }}
            >
                {children}
            </div>
        </>
    )
}

export default ProductsLayout