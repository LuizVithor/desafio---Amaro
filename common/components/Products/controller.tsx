export interface product {
    id: number;
    meta?: Meta;
    sku?: string;
    title: string;
    stock: number;
    price: number;
    rating: number;
    tags?: string[];
    brand?: string;
    weight: number;
    category: string;
    images?: string[];
    reviews?: Review[];
    thumbnail?: string;
    description: string;
    returnPolicy?: string;
    dimensions: Dimensions;
    discountPercentage: number;
    warrantyInformation?: string;
    shippingInformation?: string;
    minimumOrderQuantity: number;
    availabilityStatus: "Out of Stock" | "In Stock" | "Low Stock" | undefined;
}

interface Meta {
    qrCode: string;
    barcode: string;
    createdAt: string;
    updatedAt: string;
}

interface Review {
    date: string;
    rating: number;
    comment: string;
    reviewerName: string;
    reviewerEmail: string;
}

interface Dimensions {
    width: number;
    depth: number;
    height: number;
}