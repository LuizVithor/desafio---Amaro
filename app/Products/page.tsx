import { instance } from "@/lib/axios";
import { responseGetProducts } from "./controller";
import Products from "@/common/components/Products";

const ProductsPage = async () => {

    const data = await instance.get<responseGetProducts>('/products')

    return (
        <Products response={data.data} />
    );
};

export default ProductsPage;
