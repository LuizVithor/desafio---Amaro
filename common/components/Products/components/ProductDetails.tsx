import {
  Typography,
  Stack,
  Divider,
  Grid,
  Chip,
  List,
  ListItem,
  ListItemText,
  Rating,
} from '@mui/material';
import { parseISO } from 'date-fns';
import Barcode from 'react-barcode';
import { product } from '../controller';

interface ProductDetailsProps {
  product: product;
}

const ProductDetails: React.FC<ProductDetailsProps> = ({ product }) => {
  return (
    <Stack spacing={2}>
      <Typography variant="subtitle1" color="textSecondary">
        {product.description}
      </Typography>
      <Divider />
      <Grid container spacing={2}>
        <Grid item xs={12} sm={6}>
          <Typography variant="h6">Informações Gerais</Typography>
          <List dense>
            <ListItem>
              <ListItemText primary="SKU" secondary={product.sku} />
            </ListItem>
            <ListItem>
              <ListItemText primary="Marca" secondary={product.brand || 'N/A'} />
            </ListItem>
            <ListItem>
              <ListItemText primary="Categoria" secondary={product.category} />
            </ListItem>
            <ListItem>
              <ListItemText primary="Peso" secondary={`${product.weight} kg`} />
            </ListItem>
            <ListItem>
              <ListItemText
                primary="Dimensões"
                secondary={`${product.dimensions.width} x ${product.dimensions.depth} x ${product.dimensions.height} cm`}
              />
            </ListItem>
            <ListItem>
              <ListItemText
                primary="Quantidade Mínima de Pedido"
                secondary={product.minimumOrderQuantity}
              />
            </ListItem>
          </List>
        </Grid>
        <Grid item xs={12} sm={6}>
          <Typography variant="h6">Detalhes Adicionais</Typography>
          <List dense>
            <ListItem>
              <ListItemText
                primary="Preço"
                secondary={`R$ ${product.price.toFixed(2)}`}
              />
            </ListItem>
            <ListItem>
              <ListItemText
                primary="Desconto"
                secondary={`${product.discountPercentage}%`}
              />
            </ListItem>
            <ListItem>
              <ListItemText
                primary="Disponibilidade"
                secondary={product.availabilityStatus}
              />
            </ListItem>
            <ListItem>
              <ListItemText primary="Estoque" secondary={product.stock} />
            </ListItem>
            <ListItem>
              <ListItemText
                primary="Política de Devolução"
                secondary={product.returnPolicy}
              />
            </ListItem>
            <ListItem>
              <ListItemText
                primary="Garantia"
                secondary={product.warrantyInformation}
              />
            </ListItem>
            <ListItem>
              <ListItemText
                primary="Informações de Envio"
                secondary={product.shippingInformation}
              />
            </ListItem>
          </List>
        </Grid>
      </Grid>
      <Divider />
      <Typography variant="h6">Tags</Typography>
      <Stack direction="row" spacing={1}>
        {product.tags?.map((tag, index) => (
          <Chip key={index} label={tag} variant="outlined" />
        ))}
      </Stack>
      <Divider />
      <Typography variant="h6">Avaliações</Typography>
      {(product.reviews?.length || 0) > 0 ? (
        product.reviews?.map((review, index) => (
          <Stack key={index} spacing={1} mb={2}>
            <Stack direction="row" spacing={1} alignItems="center">
              <Typography variant="subtitle2">
                {review.reviewerName}
              </Typography>
              <Typography variant="body2" color="textSecondary">
                ({parseISO(review.date).toLocaleDateString()})
              </Typography>
            </Stack>
            <Rating value={review.rating} readOnly size="small" />
            <Typography variant="body2">{review.comment}</Typography>
          </Stack>
        ))
      ) : (
        <Typography variant="body2" color="textSecondary">
          Nenhuma avaliação disponível.
        </Typography>
      )}
      <Divider />
      <Typography variant="h6">Meta Informações</Typography>
      <List dense>
        <ListItem>
          <ListItemText primary="Código QR" />
        </ListItem>
        <Stack alignItems="start" mb={2} pl={2}>
          {product.meta?.qrCode ? (
            <img
              src={product.meta.qrCode}
              alt="Código QR"
              style={{ width: '128px', height: '128px' }}
            />
          ) : (
            <Typography variant="body2" color="textSecondary">
              Código QR não disponível.
            </Typography>
          )}
        </Stack>
        <ListItem>
          <ListItemText primary="Código de Barras" />
        </ListItem>
        <Stack alignItems="start" mb={2} pl={2} data-testid={"barcode"}>
          {product.meta?.barcode ? (
            <Barcode data-testid={"barcode"} value={product.meta.barcode} height={80} />
          ) : (
            <Typography variant="body2" color="textSecondary">
              Código de barras não disponível.
            </Typography>
          )}
        </Stack>
        <ListItem>
          <ListItemText
            primary="Criado em"
            secondary={
              product.meta?.createdAt
                ? parseISO(product.meta.createdAt).toLocaleDateString()
                : new Date().toLocaleDateString()
            }
          />
        </ListItem>
        <ListItem>
          <ListItemText
            primary="Atualizado em"
            secondary={
              product.meta?.updatedAt
                ? parseISO(product.meta.updatedAt).toLocaleDateString()
                : new Date().toLocaleDateString()
            }
          />
        </ListItem>
      </List>
    </Stack>
  );
};

export default ProductDetails;
