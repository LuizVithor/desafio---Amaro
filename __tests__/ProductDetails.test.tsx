import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { product } from '@/common/components/Products/controller';
import ProductDetails from '@/common/components/Products/components/ProductDetails';

describe('ProductDetails Component', () => {
    let sampleProduct: product;

    beforeEach(() => {
        sampleProduct = {
            id: 1,
            title: 'Produto Exemplo',
            description: 'Descrição do produto exemplo.',
            price: 99.99,
            discountPercentage: 10,
            rating: 4.5,
            stock: 10,
            brand: 'Marca Exemplo',
            category: 'Categoria Exemplo',
            thumbnail: '/images/produto-exemplo.jpg',
            images: ['/images/produto-exemplo.jpg'],
            tags: ['Exemplo', 'Teste'],
            availabilityStatus: 'In Stock',
            weight: 1.5,
            dimensions: { width: 10, depth: 5, height: 2 },
            minimumOrderQuantity: 1,
            returnPolicy: '30 dias',
            warrantyInformation: '1 ano',
            shippingInformation: 'Enviado em 2 dias',
            reviews: [
                {
                    rating: 5,
                    reviewerName: 'João',
                    date: new Date().toISOString(),
                    comment: 'Excelente produto!',
                    reviewerEmail: 'joao@gmail.com'
                },
            ],
            sku: 'SKU123',
            meta: {
                qrCode: '/images/qrcode.png',
                barcode: '123456789012',
                createdAt: new Date().toISOString(),
                updatedAt: new Date().toISOString(),
            },
        };
    });

    const renderComponent = (productProp: product) =>
        render(<ProductDetails product={productProp} />);

    it('deve renderizar a descrição do produto', () => {
        renderComponent(sampleProduct);
        expect(screen.getByText(sampleProduct.description)).toBeInTheDocument();
    });

    it('deve renderizar as informações gerais', () => {
        renderComponent(sampleProduct);
        expect(screen.getByText('Informações Gerais')).toBeInTheDocument();
        expect(screen.getByText('SKU')).toBeInTheDocument();
        expect(screen.getByText(sampleProduct.sku || "")).toBeInTheDocument();
        expect(screen.getByText('Marca')).toBeInTheDocument();
        expect(screen.getByText(sampleProduct.brand || "")).toBeInTheDocument();
        expect(screen.getByText('Categoria')).toBeInTheDocument();
        expect(screen.getByText(sampleProduct.category)).toBeInTheDocument();
        expect(screen.getByText('Peso')).toBeInTheDocument();
        expect(screen.getByText(`${sampleProduct.weight} kg`)).toBeInTheDocument();
        expect(screen.getByText('Dimensões')).toBeInTheDocument();
        expect(
            screen.getByText(
                `${sampleProduct.dimensions.width} x ${sampleProduct.dimensions.depth} x ${sampleProduct.dimensions.height} cm`
            )
        ).toBeInTheDocument();
        expect(screen.getByText('Quantidade Mínima de Pedido')).toBeInTheDocument();
        expect(
            screen.getByText(sampleProduct.minimumOrderQuantity.toString())
        ).toBeInTheDocument();
    });

    it('deve renderizar os detalhes adicionais', () => {
        renderComponent(sampleProduct);
        expect(screen.getByText('Detalhes Adicionais')).toBeInTheDocument();
        expect(screen.getByText('Preço')).toBeInTheDocument();
        expect(
            screen.getByText(`R$ ${sampleProduct.price.toFixed(2)}`)
        ).toBeInTheDocument();
        expect(screen.getByText('Desconto')).toBeInTheDocument();
        expect(
            screen.getByText(`${sampleProduct.discountPercentage}%`)
        ).toBeInTheDocument();
        expect(screen.getByText('Disponibilidade')).toBeInTheDocument();
        expect(
            screen.getByText(sampleProduct.availabilityStatus || "")
        ).toBeInTheDocument();
        expect(screen.getByText('Estoque')).toBeInTheDocument();
        expect(screen.getByText(sampleProduct.stock.toString())).toBeInTheDocument();
        expect(screen.getByText('Política de Devolução')).toBeInTheDocument();
        expect(
            screen.getByText(sampleProduct.returnPolicy || "")
        ).toBeInTheDocument();
        expect(screen.getByText('Garantia')).toBeInTheDocument();
        expect(
            screen.getByText(sampleProduct.warrantyInformation || "")
        ).toBeInTheDocument();
        expect(screen.getByText('Informações de Envio')).toBeInTheDocument();
        expect(
            screen.getByText(sampleProduct.shippingInformation || "")
        ).toBeInTheDocument();
    });

    it('deve renderizar as tags', () => {
        renderComponent(sampleProduct);
        expect(screen.getByText('Tags')).toBeInTheDocument();
        sampleProduct.tags?.forEach((tag) => {
            expect(screen.getByText(tag)).toBeInTheDocument();
        });
    });

    it('deve renderizar as avaliações', () => {
        renderComponent(sampleProduct);
        expect(screen.getByText('Avaliações')).toBeInTheDocument();
        sampleProduct.reviews?.forEach((review) => {
            expect(screen.getByText(review.reviewerName)).toBeInTheDocument();
            expect(screen.getByText(review.comment)).toBeInTheDocument();
        });
    });

    it('deve renderizar uma mensagem quando não há avaliações', () => {
        sampleProduct.reviews = [];
        renderComponent(sampleProduct);
        expect(screen.getByText('Avaliações')).toBeInTheDocument();
        expect(
            screen.getByText('Nenhuma avaliação disponível.')
        ).toBeInTheDocument();
    });

    it('deve renderizar as meta informações', () => {
        renderComponent(sampleProduct);
        expect(screen.getByText('Meta Informações')).toBeInTheDocument();
        expect(screen.getByText('Código QR')).toBeInTheDocument();
        const qrCodeImage = screen.getByAltText('Código QR');
        expect(qrCodeImage).toBeInTheDocument();
        expect(qrCodeImage).toHaveAttribute('src', sampleProduct.meta?.qrCode);

        expect(screen.getByText('Código de Barras')).toBeInTheDocument();
        const barcodeElement = screen.getByTestId('barcode');
        expect(barcodeElement).toBeInTheDocument();

        expect(screen.getByText('Criado em')).toBeInTheDocument();
        expect(screen.getByText(/Criado em/)).toBeInTheDocument();
        expect(screen.getByText('Atualizado em')).toBeInTheDocument();
    });

    it('deve renderizar uma mensagem quando o QR code não está disponível', () => {
        sampleProduct.meta!.qrCode = '';
        renderComponent(sampleProduct);
        expect(
            screen.getByText('Código QR não disponível.')
        ).toBeInTheDocument();
    });

    it('deve renderizar uma mensagem quando o código de barras não está disponível', () => {
        sampleProduct.meta!.barcode = '';
        renderComponent(sampleProduct);
        expect(
            screen.getByText('Código de barras não disponível.')
        ).toBeInTheDocument();
    });

    it('deve lidar corretamente com campos opcionais ausentes', () => {
        const { brand, ...productWithoutBrand } = sampleProduct;
        renderComponent(productWithoutBrand as product);
        expect(screen.getByText('Marca')).toBeInTheDocument();
        expect(screen.getByText('N/A')).toBeInTheDocument();
    });

    it('deve corresponder ao snapshot', () => {
        const { asFragment } = renderComponent(sampleProduct);
        expect(asFragment()).toMatchSnapshot();
    });
});
