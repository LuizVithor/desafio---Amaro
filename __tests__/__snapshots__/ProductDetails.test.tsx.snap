// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetails Component deve corresponder ao snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-1sazv7p-MuiStack-root"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle1 css-12ky0r8-MuiTypography-root"
    >
      Descrição do produto exemplo.
    </h6>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
    />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
        >
          Informações Gerais
        </h6>
        <ul
          class="MuiList-root MuiList-padding MuiList-dense css-b7dj7n-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
          >
            <div
              class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
              >
                SKU
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
              >
                SKU123
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
          >
            <div
              class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
              >
                Marca
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
              >
                Marca Exemplo
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
          >
            <div
              class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
              >
                Categoria
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
              >
                Categoria Exemplo
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
          >
            <div
              class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
              >
                Peso
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
              >
                1.5 kg
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
          >
            <div
              class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
              >
                Dimensões
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
              >
                10 x 5 x 2 cm
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
          >
            <div
              class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
              >
                Quantidade Mínima de Pedido
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
              >
                1
              </p>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
        >
          Detalhes Adicionais
        </h6>
        <ul
          class="MuiList-root MuiList-padding MuiList-dense css-b7dj7n-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
          >
            <div
              class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
              >
                Preço
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
              >
                R$ 99.99
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
          >
            <div
              class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
              >
                Desconto
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
              >
                10%
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
          >
            <div
              class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
              >
                Disponibilidade
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
              >
                In Stock
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
          >
            <div
              class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
              >
                Estoque
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
              >
                10
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
          >
            <div
              class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
              >
                Política de Devolução
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
              >
                30 dias
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
          >
            <div
              class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
              >
                Garantia
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
              >
                1 ano
              </p>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
          >
            <div
              class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
              >
                Informações de Envio
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
              >
                Enviado em 2 dias
              </p>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
    />
    <h6
      class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
    >
      Tags
    </h6>
    <div
      class="MuiStack-root css-niqf4j-MuiStack-root"
    >
      <div
        class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1lrxra9-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
        >
          Exemplo
        </span>
      </div>
      <div
        class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1lrxra9-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
        >
          Teste
        </span>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
    />
    <h6
      class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
    >
      Avaliações
    </h6>
    <div
      class="MuiStack-root css-1pxjrel-MuiStack-root"
    >
      <div
        class="MuiStack-root css-kdc3n8-MuiStack-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-1ypmsg6-MuiTypography-root"
        >
          João
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1a1whku-MuiTypography-root"
        >
          (11/11/2024)
        </p>
      </div>
      <span
        aria-label="5 Stars"
        class="MuiRating-root MuiRating-sizeSmall Mui-readOnly MuiRating-readOnly css-mrxd2j-MuiRating-root"
        role="img"
      >
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled css-1sagf1d-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
              data-testid="StarIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled css-1sagf1d-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
              data-testid="StarIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled css-1sagf1d-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
              data-testid="StarIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled css-1sagf1d-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
              data-testid="StarIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled css-1sagf1d-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
              data-testid="StarIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <p
        class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
      >
        Excelente produto!
      </p>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
    />
    <h6
      class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
    >
      Meta Informações
    </h6>
    <ul
      class="MuiList-root MuiList-padding MuiList-dense css-b7dj7n-MuiList-root"
    >
      <li
        class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root MuiListItemText-dense css-cfq8qh-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
          >
            Código QR
          </span>
        </div>
      </li>
      <div
        class="MuiStack-root css-otlaf8-MuiStack-root"
      >
        <img
          alt="Código QR"
          src="/images/qrcode.png"
          style="width: 128px; height: 128px;"
        />
      </div>
      <li
        class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root MuiListItemText-dense css-cfq8qh-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
          >
            Código de Barras
          </span>
        </div>
      </li>
      <div
        class="MuiStack-root css-otlaf8-MuiStack-root"
        data-testid="barcode"
      >
        <svg
          class=""
        />
      </div>
      <li
        class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
          >
            Criado em
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
          >
            11/11/2024
          </p>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
          >
            Atualizado em
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
          >
            11/11/2024
          </p>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
